File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        
        Dictionary<int, int> count = new Dictionary<int, int>();
 
        //Build a frequency map
        // e.g. nums=[1,1,1,2,2,3] → {1:3, 2:2, 3:1}
        foreach(int num in nums) {
            if(count.ContainsKey(num)) count[num]++;
            else count[num] = 1;
        }
 
        //Convert to a list of [frequency, number] pairs
        // → [[3,1], [2,2], [1,3]]
        List<int[]> arr = count.Select(entry => 
        new int[]{entry.Value, entry.Key}).ToList();

        //Sort by frequency descending
        // → [[3,1], [2,2], [1,3]]
        arr.Sort((a,b) => b[0].CompareTo(a[0]));

        //Take the top k numbers
        int[] res = new int[k];
        for(int i=0; i<k; i++){
            res[i] = arr[i][1];
        }
        return res;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    threeSum(nums) {

        const sorted = nums.sort((a,b) => a - b);

        let res = [];

        if (sorted[0] > 0) return [];

        for (let i = 0; i < sorted.length - 2; i++){

            let j = i + 1;
            let k = sorted.length - 1;

            if (sorted[i - 1] == sorted[i]) continue;

            while (j < k) {
                let sum = sorted[i] + sorted[j] + sorted[k];
                if (sum == 0) {
                  res.push([sorted[i],sorted[j],sorted[k]]);
                  j++;
                  k--;
                } else if (sum < 0){
                  j++;
                } else {
                  k--;
                }           
            }
            
        }
        return res;
    }
}